sky/packages/sky/lib/src/fn3/dismissable.dart

```diff
@@ -0,0 +1,257 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:sky' as sky;
+
+import 'package:sky/animation.dart';
+import 'package:sky/src/fn3/basic.dart';
+import 'package:sky/src/fn3/transitions.dart';
+import 'package:sky/src/fn3/framework.dart';
+import 'package:sky/src/fn3/gesture_detector.dart';
+
+const Duration _kCardDismissFadeout = const Duration(milliseconds: 200);
+const Duration _kCardDismissResize = const Duration(milliseconds: 300);
+final Interval _kCardDismissResizeInterval = new Interval(0.4, 1.0);
+const double _kMinFlingVelocity = 700.0;
+const double _kMinFlingVelocityDelta = 400.0;
+const double _kFlingVelocityScale = 1.0 / 300.0;
+const double _kDismissCardThreshold = 0.4;
+
+enum DismissDirection {
+  vertical,
+  horizontal,
+  left,
+  right,
+  up,
+  down
+}
+
+typedef void ResizedCallback();
+typedef void DismissedCallback();
+
+class Dismissable extends StatefulComponent {
+  Dismissable({
+    Key key,
+    this.child,
+    this.onResized,
```

Comment:
rename all the onResized stuff to be "want resize" or something along those lines. It's asking to be resized, not actually resizing itself, at least in the case of being in a MixedViewport.